src/db/spanner/models.rs

```diff
@@ -976,6 +977,48 @@ impl SpannerDb {
             .execute(&self.conn)
     }
 
+    pub fn encode_next_offset(
+        &self,
+        sort: Sorting,
+        offset: i64,
+        newer_eq: Option<i64>,
+        older_eq: Option<i64>,
+        modifieds: Vec<i64>,
+    ) -> Option<String> {
+        let mut calc_offset = 1;
+        let mut i = (modifieds.len() as i64) - 2;
+
+        let prev_bound = match sort {
+            Sorting::Index => {
+                // Use a simple numeric offset for sortindex ordering.
+                return Some(
+                    Offset {
+                        offset: offset + modifieds.len() as i64,
+                        timestamp: None,
+                    }
+                    .to_string(),
+                );
+            }
+            Sorting::None => newer_eq.unwrap_or(0),
```

Comment:
If I'm reading the python right, only Oldest uses `newer_eq`? So `None | Newest` would be `older_eq`?

Reply:
Yep, guess I read it backwards. Thanks.